writer_go18.go

```diff
@@ -0,0 +1,211 @@
+//go:build go1.18
+
+package parquet
+
+import (
+	"io"
+	"reflect"
+)
+
+// GenericWriter is similar to a Writer but uses a type parameter to define the
+// Go type representing the schema of rows being written.
+//
+// Using this type over Writer has multiple advantages:
+//
+// - By leveraging type information, the Go compiler can provide greater
+//   guarantees that the code is correct. For example, the parquet.Writer.Write
+//   method accepts an argument of type interface{}, which delays type checking
+//   at the runtime. The parquet.GenericWriter[T].Write method ensures at
+//   compile time that the values it receives will be of type T, reducing the
+//   risk of introducing bugs.
```

Comment:
```suggestion
//   risk of introducing errors.
```

Nit, I don't like "bugs" which implies something don't have much control over like an insect flying into your room.